dht/partition_filter.hh

```diff
@@ -24,7 +24,7 @@ public:
     }
 
     // Must be called with increasing token values.
-    bool belongs_to_current_node(const dht::token& t) {
+    bool belongs_to_current_node(const dht::token& t) const noexcept {
```

Comment:
I checked and currently this method is indeed noexcept but this is in no way guaranteed if either `itnerval::contains()` or `interval::after()` starts to allocate. The former indeed can allocate under certain circumstances.
This is why I don't like automatically adding `noexcept` to "simple" methods. I'm sure you checked that it is fine to add, but allocations can hide deep in other simple methods and can cause unpleasant surprises later.

Reply:
I think the interval predicates shouldn't allocate either, by contract...

Reply:
According to what contract?

Reply:
a contract by declaration as noexcept (i.e. promising that no exception will escape the call)

Reply:
Right, but this is not how contracts work. Slapping `noexcept` onto a top-level method doesn't mean that now all methods it calls have to be `noexcept` by definition too.
Each method has to be examined individually. And when it comes to methods that call other methods: they can be `noexcept` if they either catch all exceptions or all methods it calls are also `noexcept`.

Reply:
Of course.  I'll send a PR that marks the low-level functions noexcept.